app/main/views/templates.py

```diff
@@ -115,11 +119,13 @@ def choose_template(service_id, template_type='all'):
 
     return render_template(
         'views/templates/choose.html',
-        templates=current_service.templates_by_type(template_type),
-        show_search_box=(len(current_service.templates_by_type(template_type)) > 7),
+        template_folder_path=current_service.get_template_folder_path(template_folder_id),
+        template_folders=current_service.get_template_folders(template_folder_id),
+        templates=current_service.get_templates(template_type, template_folder_id),
+        show_search_box=(len(current_service.get_templates(template_type)) > 7),
```

Comment:
This should depend on the number of visible templates I think, so
```suggestion
        show_search_box=(len(current_service.get_templates(template_type, template_folder_id)) > 7),
```

Reply:
I haven't touched the search box at the moment, but I think it's more complex than that if we have a lot of templates in nested folders and want the search to show them. Right now they're not added to the page at all, we'll need to come back to this once we can actually put templates inside folders.